src/parseTools.js

```diff
@@ -1335,8 +1335,57 @@ function from64(x) {
 }
 
 function to64(x) {
-  if (!MEMORY64) return x;
-  return `BigInt(${x})`;
+  if (MEMORY64) return /^\d+$/.test(x) ? `${x}n` : `BigInt(${x})`;
```

Comment:
I'm not sure we can do this atm. The `10n` notation requires a newer version of terser than we have here, last I checked. I verified that it still breaks with `-O3` on this:
```cpp
#include <emscripten.h>

int main() {
  EM_ASM({
    Module['foo'](10n);
  });
```
That crashes in terser.

Reply:
oh, sad. :( Would updating be possible? If not, we could as a temporary workaround unconditionally use `BigInt()` constructor, and once we have an update in place, switch to supporting the literals?

Reply:
For now, can write `BigInt(10)`, yeah. That's what we've done elsewhere.

Eventually upgrading terser would be good, but I'm not sure how easy that would be.